core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -949,7 +949,15 @@ public void unparseOffsetFetch(SqlWriter writer, SqlNode offset,
    * @param offset Number of rows to skip before emitting, or null
    * @param fetch Number of rows to fetch, or null
    */
-  public void unparseTopN(SqlWriter writer, SqlNode offset, SqlNode fetch) {
+  public void unparseFetchAsTopN(SqlWriter writer, SqlNode offset, SqlNode fetch) {
+  }
+
+  /**
+   * Unparses a TOP N clause (ex. SELECT TOP 5 * FROM FOO)
+   * @param topN
+   */
+  public void unparseTopN(SqlWriter writer, SqlNode topN) {
```

Comment:
Should we check for a null topN here?

Reply:
This check is done in SqlSelectOperator, but the unparseTopN could theoretically be called from somewhere else as well, so it probably does make sense to add it just to be safe.